src/Catalyst.Cli/CommandTypes/BaseMessageCommand.cs

```diff
@@ -0,0 +1,147 @@
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Concurrent;
+using System.Reactive.Linq;
+using Catalyst.Abstractions.Cli.Commands;
+using Catalyst.Abstractions.Cli.CommandTypes;
+using Catalyst.Abstractions.Cli.Options;
+using Catalyst.Abstractions.Rpc;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.IO.Events;
+using Catalyst.Core.Lib.IO.Messaging.Dto;
+using Catalyst.Protocol.Peer;
+using Google.Protobuf;
+using Serilog;
+
+namespace Catalyst.Cli.CommandTypes
+{
+    public abstract class BaseMessageCommand<TRequest, TResponse, TOption> : BaseCommand<TOption>,
+        IMessageCommand<TRequest>, IDisposable
+        where TRequest : IMessage<TRequest>
+        where TResponse : IMessage<TResponse>
+        where TOption : IOptionsBase
+    {
+        private readonly IDisposable _eventStreamObserverClientAdded;
+        private readonly IDisposable _eventStreamObserverClientRemoved;
+        private readonly ConcurrentDictionary<int, IDisposable> _subscriptions;
+        private PeerId _recipientPeerId;
+
+        protected BaseMessageCommand(ICommandContext commandContext, ILogger logger)
+            : base(commandContext, logger)
+        {
+            _subscriptions = new ConcurrentDictionary<int, IDisposable>();
+            _eventStreamObserverClientAdded = CommandContext.SocketClientRegistry.EventStream
+               .OfType<SocketClientRegistryClientAdded>().Subscribe(SocketClientRegistryClientAddedOnNext);
+            _eventStreamObserverClientRemoved = CommandContext.SocketClientRegistry.EventStream
+               .OfType<SocketClientRegistryClientRemoved>().Subscribe(SocketClientRegistryClientRemovedOnNext);
+        }
+
+        protected PeerId RecipientPeerId
+        {
+            get
+            {
+                if (_recipientPeerId != null) return _recipientPeerId;
+                var rpcClientConfig = CommandContext.GetNodeConfig(Options.Node);
+                _recipientPeerId =
+                    rpcClientConfig.PublicKey.BuildPeerIdFromBase32Key(rpcClientConfig.HostAddress,
+                        rpcClientConfig.Port);
+                return _recipientPeerId;
+            }
+        }
+
+        protected PeerId SenderPeerId => CommandContext.PeerId;
+
+        public void Dispose() { Dispose(true); }
+
+        public IRpcClient Target => CommandContext.GetConnectedNode(Options.Node);
+
+        public virtual void SendMessage(TOption options)
+        {
+            var message = GetMessage(options);
+
+            if (message == null) return;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)